src/openfl/display/DisplayObject.hx

```diff
@@ -1284,9 +1284,6 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 			
 			if (updateTransform) {
 				
-				__update (false, true);
-				
-				Matrix.__pool.release (matrix);
 				Rectangle.__pool.release (rect);
```

Comment:
Like discussed, I think the rect needs to be released on (updateTransform || needRender)